sklearn/cluster/hierarchical.py

```diff
@@ -365,6 +365,10 @@ def linkage_tree(X, connectivity=None, n_components=None,
     --------
     ward_tree : hierarchical clustering with ward linkage
     """
+    if n_components is not None:
```

Comment:
To be sure, we should really be testing against a different value, like `if n_components != 'deprecated'` and setting the default to `'deprecated'` so that a user explicitly passing None will receive the warning.